lib/spack/spack/spec_parser.py

```diff
@@ -189,6 +200,16 @@ def accept(self, kind: SpecTokens):
     def expect(self, *kinds: SpecTokens):
         return self.next_token and self.next_token.kind in kinds
 
+    def push(self, token_stream: Iterator[Token]):
```

Comment:
Hmmm, this seems very coupled with the global CONFIG Wondering if we could keep the structure more modular.